src/coreclr/vm/interpexec.cpp

```diff
@@ -1115,6 +1282,8 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
     }
 
 EXIT_FRAME:
+
+    frame_data_pop_info(&pThreadContext->frameDataAllocator, (InterpreterFrame*)pFrame);
```

Comment:
On mono we have separate return opcode so we don't do this on the main return path. Please add a FIXME so we address it in the future.

Reply:
Added, thanks. Currently, it is guarded with `pInfos[top].frame == pFrame`.